lib/client.js

```diff
@@ -140,12 +166,20 @@ Client.prototype._request = function(options, callback) {
   options.method = "POST";
   options.headers = { "User-Agent": this.userAgent };
 
+  var _this = this;
   return new Promise(function(resolve, reject) {
     request(options, function (error, response) {
-      if (!error && response.statusCode >= 400)
-        error = Error("Heap API server error " + response.statusCode);
+      if (error) {
+        response = null;
+      } else {
+        if (response.statusCode >= 400) {
+          error = Error("Heap API server error " + response.statusCode + ": " +
```

Comment:
perhaps attach the response object to the error? I think node-pg does something like this for some pg errors for an awesome effect.